test/es-module/test-esm-import-meta-main-eval.mjs

```diff
@@ -0,0 +1,37 @@
+import { escapePOSIXShell, mustSucceed } from '../common/index.mjs';
+import child from 'node:child_process';
+
+const execOptions = '--input-type module';
+
+const script = `
+import { mustCall } from '../common/index.mjs';
+import assert from 'node:assert/strict';
+
+assert.strictEqual(import.meta.main, true);
+
+import('../fixtures/es-modules/import-meta-main.mjs').then(
+  mustCall(({ isMain }) => {
+    assert.strictEqual(isMain, false);
+  })
+);
+`;
+
+const scriptInWorker = `
+import { Worker } from 'node:worker_threads';
+new Worker(\`${script}\`, { eval: true });
+`;
+
+function test_evaluated_script() {
+  const [cmd, opts] = escapePOSIXShell`"${process.execPath}" ${execOptions} --eval "${script}"`;
```

Comment:
Please don't use `escapePOSIXShell`, it's meant for tests that have no other choice than using a shell, but there's no reason to use it, let's use `spawnPromisified` or `spawnSyncAndAssert`

Reply:
It seems I've inadvertently invoked Cunningham's law :joy: